### PR TITLE
Improve wording of “Connecting Applications with Services”

### PR DESCRIPTION
**Pull request by tom1299**
The meaning of original sentence: "Coordinating ports across multiple **developers** is very difficult to do at scale..." is not really obvious. From what I understand from the context of this page, the term **containers** would make more sense, because the context in this paragraph is on containers and ports rather than developers.

>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.17 Features: set Milestone to 1.17 and Base Branch to dev-1.17
>
> For pull requests on Chinese localization, set Base Branch to release-1.16
> Feel free to ask questions in #kubernetes-docs-zh
>
> For pull requests on Korean Localization: set Base Branch to dev-1.16-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by sftim** [comment]
The existing document is trying to say that if multiple people need to select a port, those people need to agree on who is using what port where, plus things like handing back ports that are no longer used.

It's more convenient, if you can, to let the control plane take care of port allocations.


I like the aim of the existing document. However, it sounds like the wording could be more clear.

**Comment by sftim** [comment]
I'd add that, with Kubernetes, picking what ports to use for the containers inside a Pod is rarely difficult. It's when you have multiple different Pod templates that things could get gnarly.

**Comment by tom1299** [comment]
Thank you very much for the explanation. From that point of view the sentence indeed makes sense as it is. I will try to change the wording a bit in order to make the intent a little clearer.

**Comment by sftim** [comment]
/retitle Improve wording of “Connecting Applications with Services”

**Comment by sftim** [comment]
@tom1299 if this is something you plan to work on, you can label it as work in progress by putting `[WIP] ` at the start of the pull request title

**Comment by tom1299** [comment]
I'm absolutely sorry for the mess I caused. But I straightened out git commits and everything should be fine now.
I rephrased the beginning of the sentence a little bit to make it more clear that the the people or teams need to agree on the ports of their containers.
What do you thing ?

File: content/en/docs/concepts/services-networking/connect-applications-service.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Now that you have a continuously running, replicated application you can expose
...
+Coordinating ports across multiple developers or teams that provide containers is very difficult to do at scale and exposes users to cluster-level issues outside of their control. Kubernetes assumes that pods can communicate with other pods, regardless of which host they land on. We give every pod its own cluster-private-IP address so you do not need to explicitly create links between pods or map container ports to host ports. This means that containers within a Pod can all reach each other's ports on localhost, and all pods in a cluster can see each other without NAT. The rest of this document will elaborate on how you can run reliable services on such a networking model.
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/connect-applications-service.md]: Here's a further, optional tweak to wording @tom1299 
```suggestion
Coordinating port allocations across multiple developers or teams that provide containers is very difficult to do at scale, and exposes users to cluster-level issues outside of their control. Kubernetes assumes that pods can communicate with other pods, regardless of which host they land on. Kubernetes gives every pod its own cluster-private IP address, so you do not need to explicitly create links between pods or map container ports to host ports. This means that containers within a Pod can all reach each other's ports on localhost, and all pods in a cluster can see each other without NAT. The rest of this document elaborates on how you can run reliable services on such a networking model.
```

**Pull request merged**